add episode on vaccine strategies (#73)
epiverse-trace/tutorials-late: episodes/vaccine-comparisons.Rmd

READ THESE LINES AND ERASE:

The Workbench-related sections that the developer must keep are:

- YAML on top
- Questions
- Objectives
- Keypoints

The Epiverse-TRACE sections that we encourage to keep are:

- Prerequisites
- Introduction

Take a look to the Contributing.md file for more writing guidelines.

::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::::

## Direct and indirect effects of vaccination

Vaccination programs using infection-blocking vaccines have two benefits:

1. Reducing individual risk of infection (**direct** effect of vaccination)
2. Reducing onward contribution to transmission (**indirect** effect of vaccination).

We will illustrate this using results from `model_default()` in `{epidemics}`. Using the contact matrix and parameters from the COVID-19 outbreak example in [Modelling interventions](../episodes/modelling-interventions.md) we will investigate the impact of two vaccination programs on the number of infections. We will assume that vaccination programs start on day 0 and continue to be in place for 150 days. We will assume all age groups are vaccinated at the same rate in each program as follows :

> + vaccination program 01 : vaccination rate 0.001

Author: I believe it is the proportion of those susceptible, [see here](
https://github.com/epiverse-trace/epidemics/blob/f706261bac19282e813ca43675ff16dcb79b3d40/R/model_default.R#L343C11-L343C21)

  # time
  time_end = 200, increment = 1.0
)

# create intervention_type column for plotting
output_baseline$intervention_type <- "baseline"
output_vaccinate_01$intervention_type <- "vaccinate 0.001"
output_vaccinate_02$intervention_type <- "vaccinate 0.01"
output <- rbind(output_baseline, output_vaccinate_01, output_vaccinate_02)

output %>%
  filter(compartment == "vaccinated") %>%
  ggplot() +
  ggtitle("Number vaccinated") +
  aes(
    x = time,
    y = value,
    color = intervention_type,
    linetype = intervention_type
  ) +
  stat_summary(
    fun = "sum",
    geom = "line",
    linewidth = 1
  ) +
  scale_y_continuous(
    labels = scales::comma
  ) +
  geom_vline(
    xintercept = c(
      vaccinate_01$time_begin,
      vaccinate_01$time_end
    ),
    linetype = 2
  ) +
  theme_bw() +
  labs(
    x = "Simulation time (days)",
    y = "Individuals"
  )
```

:::

To show the **indirect** effect of vaccinations, we want to see the effect that vaccination has on the number of new infections. We will use the function `new_infections()` in `{epidemics}` to calculate the number of new infections over time for the different vaccination programs. 

The inputs required are :

> + `data` : the model output,

Reviewer: Beyond scope of this specific PR, but wonder if it's unclear to learners to have epidemics models generate an output labelled 'data' for simulations?

                                          by_group = FALSE)
vaccinate_02_infections <- new_infections(output_vaccinate_02,
                                          compartments_from_susceptible =
                                            "vaccinated",
                                          by_group = FALSE)

# create intervention_type column for plotting
baseline_infections$intervention_type <- "baseline"
vaccinate_01_infections$intervention_type  <- "vaccinate 0.001"
vaccinate_02_infections$intervention_type  <- "vaccinate 0.01"
infections <- rbind(baseline_infections,
                    vaccinate_01_infections,
                    vaccinate_02_infections)

infections %>%
  ggplot() +
  ggtitle("New infections") +
  geom_line(
            aes(time, new_infections, colour = intervention_type,
              linetype = intervention_type
            ),
            linewidth = 1) +
  scale_y_continuous(
    labels = scales::comma
  ) +
  geom_vline(
    xintercept = c(
      vaccinate_01$time_begin,
      vaccinate_01$time_end
    ),
    linetype = 2
  ) +
  theme_bw() +
  labs(
    x = "Simulation time (days)",
    y = "Individuals"

>   )

Reviewer: Would it be useful to also output a quick table comparing cumulative infections under each scenario?

This **indirect** effect of vaccination programs is key to successfully controlling and eradicating infections. 

::::::::::::::::::::::::::::::::::::: callout
## Herd immunity threshold

For homogeneous mixing populations, there exists a target threshold of vaccination based on the basic reproduction number $R_0$ to achieve [herd immunity](../learners/reference.md#herdimmunity) .

The proportion of the population ($p$) that needs to be immune to achieve herd immunity is:

$$p = 1- \frac{1}{R_0}. $$

For details of the mathematical derivation check out this [Plus Magazine article](https://plus.maths.org/content/maths-minute-r0-and-herd-immunity).

::::::::::::::::::::::::::::::::::::::::::::::::

## Targeted vaccination

For infections that have a higher burden in different risk groups, targeted vaccination programs can be utilised to control infection. For example, if there is heterogeneity in contacts by age, targeting different age groups for vaccination will result in different disease trajectories. 

> The contact matrix for our example is :

Reviewer: Reiterate it's the same matrix as one loaded above?

```{r}
contact_matrix
```

There is higher levels of mixing within the 0-15 and 15-65 age group, than in the 65+ age a group so we expect that targeting different age groups will result in different disease trajectories. 

To show the effect of targeted vaccination, we will compare the following scenarios:

> + vaccinating each age group at a rate 0.001,

Reviewer: ```suggestion
+ vaccinating all age groups at a rate 0.001,
```

vaccinate_group_3_infections <- new_infections(output_vaccinate_group_3,
                                               compartments_from_susceptible =
                                                 "vaccinated", by_group = FALSE)

vaccinate_group_1_infections$intervention_type  <- "vaccinate group 1"
vaccinate_group_2_infections$intervention_type  <- "vaccinate group 2"
vaccinate_group_3_infections$intervention_type  <- "vaccinate group 3"

output_infections <- rbind(baseline_infections,
                           vaccinate_01_infections,
                           vaccinate_group_1_infections,
                           vaccinate_group_2_infections,
                           vaccinate_group_3_infections)

output_infections %>%
  ggplot() +
  geom_line(
            aes(time, new_infections, colour = intervention_type,
              linetype = intervention_type
            ),
            linewidth = 1) +
  scale_y_continuous(
    labels = scales::comma
  ) +
  geom_vline(
    xintercept = c(
      vaccinate_01$time_begin,
      vaccinate_01$time_end
    ),
    linetype = 2
  ) +
  theme_bw() +
  labs(
    x = "Simulation time (days)",
    y = "Individuals"

>   )

Reviewer: As above, a table with cumulative infections here would be useful to show overall differences

```

:::

> Vaccinating group 3 only results in the highest epidemic peak size. Targeting group 2 results in the smallest epidemic peak size. Whereas targeting all age groups pushes the epidemic peak time later.  

Reviewer: Would it be useful to have a short subsection after this one that implements an age specific infection-fatality-risk? E.g. take the age-stratified cumulative infections for each scenario, then scale by age-specific risk? This could be a nice challenge to show that targeting younger groups can reduce impact of infections that have higher IFR for elderly?

Author: I've added a short section on this using the same contact matrix as before. The example does show that vaccinating other age groups has an effect on deaths in the eldest group, but not the youngest age group. The example could be updated with a different contact matrix to illustrate that point more heavily if wanted.

Reviewer: Thanks, looks good to me

> Vaccinating group 3 only results in the highest epidemic peak size. Targeting group 2 results in the smallest epidemic peak size. Whereas targeting all age groups pushes the epidemic peak time later.  

Reviewer: Worth briefly mentioning (either here or referencing elsewhere if already covered in another episode) that we're often interested in both the peak size (i.e. healthcare pressure at single point in time) as well as overall epidemic size (i.e. cumulative disease)?

## Vaccination versus NPIs

Modelling is a useful tool for understanding the potential impact of the timing and different combinations of control measures. In an outbreak scenario, vaccines can be used alongside other control measures. A recent study by [Barnsley et al. (2024)](https://doi.org/10.1016/S2214-109X(24)00286-9) used modelling to show the impact the COVID-19 vaccine could have had alongside NPIs if the vaccine had been developed within 100 days of the pathogen threat being recognised. We will use this study as inspiration to show the impact of a vaccine versus NPI implemented at different times. 

> The NPI we will consider is closing schools. Specifically, a school closure that will reduce the contacts between school aged children (aged 0-15) by 0.5, and will cause a small reduction (0.01) in the contacts between adults (aged 15 and over). For the vaccine program, we assume an equal vaccination rate of 0.01 in each age group.

Reviewer: ```suggestion
The NPI we will consider is temporarily closing schools. Specifically, a school closure that will reduce the contacts between school aged children (aged 0-15) by 0.5, and will cause a small reduction (0.01) in the contacts between adults (aged 15 and over). For the vaccine program, we assume an equal vaccination rate of 0.01 in each age group.
```
Maybe add a citation for school closures, which is common proposed measure for pandemic flu - and sometimes seasonal flu also: https://pmc.ncbi.nlm.nih.gov/articles/PMC2609897/. The damage of closures for COVID is likely to make school closure unappealing as policy option for future pandemic, but for a severe infection among younger groups, would likely end up on the cards (whether explicitly or by default, via behaviour change).

> We define **early implementation** as the first day of the simulation, and **late implementation** as 50 days after the start of the simulation. We assume the control measures are in place for 100 days after they start. 

Reviewer: Maybe say 'first day of the simulation (i.e. when there are X infections in total)'? And similar for 50 days - as this is closer to the sort of triggers that would be used in real-life?

  population = uk_population,
  transmission_rate = transmission_rate,
  infectiousness_rate = infectiousness_rate,
  recovery_rate = recovery_rate,
  intervention = list(contacts = close_schools_early),
  time_end = time_end, increment = 1.0
)

# vaccination only (early)
output_vacc_early <- model_default(
  population = uk_population,
  transmission_rate = transmission_rate,
  infectiousness_rate = infectiousness_rate,
  recovery_rate = recovery_rate,
  vaccination = vacc_early,
  time_end = time_end, increment = 1.0
)

# vaccination only (late)
output_vacc_late <- model_default(
  population = uk_population,
  transmission_rate = transmission_rate,
  infectiousness_rate = infectiousness_rate,
  recovery_rate = recovery_rate,
  vaccination = vacc_late,
  time_end = time_end, increment = 1.0
)

output_npi_early_vacc_late$intervention_type <- "early npi, late vacc"
output_npi_early$intervention_type <- "early npi"
output_vacc_early$intervention_type <- "early vacc"
output_vacc_late$intervention_type <- "late vacc"

output <- rbind(output_baseline, output_vacc_late,
                output_vacc_early, output_npi_early,
                output_npi_early_vacc_late)

```

> Early implementation of NPIs (`early npi`) shifts the timing of the peak of infectious individuals but does not decrease the magnitude by much. Early implementation of the vaccine (`early vacc`) is the most effective control for reducing the peak number of infectious individuals. Having an NPI in place before the vaccine is implemented (`early npi, late vacc`) is more effective than simply just implementing the vaccine late (`late vacc`).

Reviewer: ```suggestion
Early implementation of NPIs (`early npi`) delays the timing of the peak of infectious individuals but does not decrease the magnitude by much once the measure is lifted. Early implementation of the vaccine (`early vacc`) is the most effective control for reducing the peak number of infectious individuals, because it reduces the size of the susceptible group rather than just temporarily keeping infections away from it. Having an NPI in place before the vaccine is implemented (`early npi, late vacc`) is more effective than simply just implementing the vaccine late (`late vacc`).
```

  vaccination = vacc_late,
  intervention = list(contacts = close_schools_early_50),
  time_end = 300, increment = 1.0
)

output_npi_early_vacc_late_20$intervention_type <- "20 days"
output_npi_early_vacc_late_50$intervention_type <- "50 days"
output_npi_early_vacc_late$intervention_type <- "100 days"

output_npis <- rbind(output_npi_early_vacc_late_20,
                     output_npi_early_vacc_late_50,
                     output_npi_early_vacc_late)

output_npis %>%
  filter(compartment == "infectious") %>%
  ggplot() +
  aes(
    x = time,
    y = value,
    color = intervention_type,
    linetype = intervention_type
  ) +
  stat_summary(
    fun = "sum",
    geom = "line",
    linewidth = 1
  ) +
  scale_y_continuous(
    labels = scales::comma
  ) +
  geom_vline(
    xintercept = c(
      early_start,
      early_start + duration
    ),
    linetype = 2
  ) +
  geom_vline(
    xintercept = c(
      late_start,
      late_start + duration
    ),
    linetype = 3
  ) +
  theme_bw() +
  labs(
    x = "Simulation time (days)",
    y = "Individuals"
  )
```

:::::::::::::::::::::::::::::::::
::::::::::::::::::::::::::::::::::::::::::::::::

## Summary

> Using model generated disease trajectories we have investigated the potential impact of different vaccination strategies. When infection has group specific risk, targeted vaccination programs may be more effective in reducing epidemic peak. NPIs can be utilised alongside vaccine roll outs to control epidemic peaks when vaccines may not be readily available. 

Reviewer: ```suggestion
Using model generated disease trajectories we have investigated the potential impact of different vaccination strategies. When infection has group specific risk, targeted vaccination programs may be more effective in reducing epidemic peak. NPIs can be used alongside vaccine development to delay epidemic peaks before vaccines are readily available. 
```

> In this tutorial we have investigated the potential effect of infection blocking vaccines, but we can use models to explore other effects of vaccines including imperfect vaccines. The disease trajectories generated in this tutorial can be used to calculate measures such as DALYs averted as part of health economic analyses.

Reviewer: Worth adding some suggested further reading for DALYs – especially if above subsection on age-specific deaths included, as could give a sentence of intuition here.

Author: Added DALYs to the glossary with a link to further reading

::::::::::::::::::::::::::::::::::::: keypoints 

> - Herd immunity is a indirect effect of vaccination programs

Reviewer: Probably beyond scope of this episode to model, but could get them to think about the impact vaccination would have if it came very late in the epidemic (i.e. after post-infection immunity had accumulated) – this happened in 2009, for example: https://pubmed.ncbi.nlm.nih.gov/20096762/
